src/mewbot/behaviour.py

```diff
@@ -75,29 +21,31 @@ def __init__(self) -> None:
         self.conditions = []
         self.actions = []
 
-    def add(self, block: BehaviourComponent) -> None:
-        if not isinstance(block, BehaviourComponent):
+    def add(
+        self, component: Union[TriggerInterface, ConditionInterface, ActionInterface]
+    ) -> None:
+        if not isinstance(component, (TriggerInterface, ConditionInterface, ActionInterface)):
             raise TypeError()
 
-        interests = block.consumes_inputs()
+        interests = component.consumes_inputs()
         interests = self.interests.intersection(interests) if self.interests else interests
 
         if not interests:
             raise ValueError("Adding a behaviour component that doesn't match input types")
 
         self.interests = interests
 
-        if isinstance(block, Trigger):
-            self.triggers.append(block)
-        if isinstance(block, Condition):
-            self.conditions.append(block)
-        if isinstance(block, Action):
-            self.actions.append(block)
+        if isinstance(component, TriggerInterface):
```

Comment:
Is there a potential use case for an induvidual component to be an instance of multiple of these? Can I have a component that is both a TriggerInterface and a ConditionInterface?
Do we want to accept those conditions, warn on those conditions, or terminate?

Reply:
Doing so is explicity forbidden in the component API (https://github.com/ajCameron/mewbot/blob/2990b63/src/mewbot/component.py#L43-L46) but nothing prevents alternative implementations of these interfaces doubling up.